src/api/appUsers.js

```diff
@@ -1,23 +1,69 @@
-import BaseAPI from './base';
-import http from '../utils/http';
+import { BaseAPI } from './base';
+import { http } from '../utils/http';
 
-export default class AppUsersAPI extends BaseAPI {
-  retrieve(userId, auth) {
-    const url = this.getFullURL('appUsers', userId);
+
+/**
+ * Init API properties
+ * @typedef InitProps
+ */
+
+/**
+ * @class AppUsersAPI
+ * @extends BaseAPI
+ */
+export class AppUsersAPI extends BaseAPI {
+
+  /**
+   * Initializes the conversation for a user
+   * @param  {InitProps} props
+   * @param  {AuthCredentials} auth
+   * @return {APIResponse}
+   */
+  init(props, auth) {
+    const url = this.getFullURL('init');
+    return this.getAuthenticationHeaders(auth).then((authHeaders) => {
```

Comment:
Could have `baseApi` wrap the util `http`/`get` methods, and then it can take care of applying any auth settings specified in `props`. It could also take care of `getFullUrl`.

This implementation then becomes:

```
return `this.post('init', props);
```